cfgov/core/utils.py

```diff
@@ -98,38 +100,46 @@ def is_image_tag(tag):
     return False
 
 
-def add_link_markup(tags):
-    for tag in tags:
-        icon = False
-        if not tag.attrs.get('class', None):
-            tag.attrs.update({'class': []})
-        if tag['href'].startswith('/external-site/?'):
-            # Sets the icon to indicate you're leaving consumerfinance.gov
-            icon = 'external-link'
-            components = urlparse(tag['href'])
-            arguments = parse_qs(components.query)
-            if 'ext_url' in arguments:
-                external_url = arguments['ext_url'][0]
-                tag['href'] = signed_redirect(external_url)
-
-        elif NONCFPB_LINK_PATTERN.match(tag['href']):
-            # Sets the icon to indicate you're leaving consumerfinance.gov
-            icon = 'external-link'
-            if EXTERNAL_LINK_PATTERN.match(tag['href']):
-                tag['href'] = signed_redirect(tag['href'])
-        elif DOWNLOAD_LINK_PATTERN.search(tag['href']):
-            # Sets the icon to indicate you're downloading a file
-            icon = 'download'
-        if icon:
-            tag.attrs['class'].append(LINK_ICON_CLASSES)
-            # Wraps the link text in a span that provides the underline
-            contents = tag.contents
-            span = BeautifulSoup('', 'html.parser').new_tag('span')
-            span['class'] = LINK_ICON_TEXT_CLASSES
-            span.contents = contents
-            tag.contents = [span, NavigableString(' ')]
-            # Appends the SVG icon
-            tag.contents.append(BeautifulSoup(svg_icon(icon), 'html.parser'))
+def add_link_markup(tag):
+    """Adds an external link icon if the input is not a CFPB (internal) link.
+    Adds an external link redirect if the input is not a gov link.
+    Adds a download icon if the input is a file.
+    Otherwise (internal link), returns None.
+    """
+    icon = False
+    if not tag.attrs.get('class', None):
+        tag.attrs.update({'class': []})
+    if tag['href'].startswith('/external-site/?'):
```

Comment:
@Scotchester There are pages with this hardcoded in it... e.g. if you look at this page http://localhost:8000/admin/pages/2982/edit/ (locally if you have a production dump, or on content) you'll see it in the content section.  I think this is the case on all legacy pages.  I'm not too familiar with the history of these pages, but it looks like they were part of some kind of Wordpress migration and at some point we wanted to add in the external link icons differently than are defined in the hardcoded URL, hence the condition here.

Reply:
I guess it may not be feasible/worth the effort to replace all those instances? The code in this PR would now add the redirect if it wasn't hard-coded, right?

Reply:
Yeah I think that's a little out of scope (by the way I didn't introduce this hardcoding, it was already a part of the function).